chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpEndpoint.java

```diff
@@ -0,0 +1,114 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * An object that represents HTTP-endpoint. It consists of {@link JHttpEndpoint#protocol},
+ * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields. <p>
+ *
+ * @author Anton Antonenko
+ * @since 1.2.7
+ */
+@SuppressWarnings("unused")
+public class JHttpEndpoint {
+
+    /**
+     * Enum representing HTTP and HTTPS protocols
+     */
+    public enum Protocol {
+        HTTP("http"), HTTPS("https");
+
+        private String value;
+
+        Protocol(String value) {
+            this.value = value;
+        }
+    }
+
+    private Protocol protocol = Protocol.HTTP;
+    private String hostname;
+    private int port;
+
+    public Protocol getProtocol() {
+        return protocol;
+    }
+
+    public void setProtocol(Protocol protocol) {
+        this.protocol = protocol;
+    }
+
+    public String getHostname() {
+        return hostname;
+    }
+
+    public void setHostname(String hostname) {
+        this.hostname = hostname;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    /**
+     * @return {@link URI} based on {@link JHttpEndpoint#protocol},
+     * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields values.
+     */
+    public URI getURI() {
+        return URI.create(String.format("%s://%s:%s", protocol.value, hostname, port));
+    }
+
+    /**
+     * @param queryParams query parameters to be added to URI
+     * @return {@link URI} based on {@link JHttpEndpoint#protocol},
+     * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields values with <b>queryParams</b> added.
+     */
+    public URI getURI(Map<String, String> queryParams) {
```

Comment:
1. "queryParams" naming might be confused with JHttpQuerry.
2. I am a little bit doubt whether we need such methods here, inside Endpoint.